test/Microsoft.AspNetCore.Sockets.Tests/HttpConnectionDispatcherTests.cs

```diff
@@ -452,6 +454,273 @@ public class HttpConnectionDispatcherTests
+            var authFeature = new HttpAuthenticationFeature();
+            authFeature.Handler = new TestAuthenticationHandler(context);
+            context.Features.Set<IHttpAuthenticationFeature>(authFeature);
+
+            // "authorize" user
+            context.User = new ClaimsPrincipal(new ClaimsIdentity(new[] { new Claim(ClaimTypes.NameIdentifier, "name") }));
+
+            var endPointTask = dispatcher.ExecuteAsync<BlockingEndPoint>("", context);
+            await state.Connection.Transport.Output.WriteAsync(new Message(Encoding.UTF8.GetBytes("Hello, World"), MessageType.Text)).OrTimeout();
+
+            await endPointTask.OrTimeout();
+
+            Assert.Equal(StatusCodes.Status200OK, context.Response.StatusCode);
+            Assert.Equal("T12:T:Hello, World;", GetContentAsString(context.Response.Body));
+        }
+
+        [Fact]
+        public async Task AuthorizedConnectionWithRejectedSchemesFailsToConnectToEndPoint()
+        {
+            var manager = CreateConnectionManager();
+            var state = manager.CreateConnection();
+            var dispatcher = new HttpConnectionDispatcher(manager, new LoggerFactory());
+            var context = new DefaultHttpContext();
+            var services = new ServiceCollection();
+            services.AddOptions();
+            services.AddEndPoint<BlockingEndPoint>(options =>
+            {
+                options.AuthorizationPolicyNames.Add("test");
+            });
+            services.AddAuthorization(options =>
+            {
+                options.AddPolicy("test", policy =>
+                {
+                    policy.RequireClaim(ClaimTypes.NameIdentifier);
+                    policy.AddAuthenticationSchemes("Default");
+                });
+            });
+            services.AddLogging();
+
+            context.RequestServices = services.BuildServiceProvider();
+            context.Request.Path = "/poll";
+            var values = new Dictionary<string, StringValues>();
+            values["id"] = state.Connection.ConnectionId;
+            var qs = new QueryCollection(values);
+            context.Request.Query = qs;
+            context.Response.Body = new MemoryStream();
+            var authFeature = new HttpAuthenticationFeature();
+            authFeature.Handler = new TestAuthenticationHandler(context, acceptScheme: false);
+            context.Features.Set<IHttpAuthenticationFeature>(authFeature);
+
+            // "authorize" user
+            context.User = new ClaimsPrincipal(new ClaimsIdentity(new[] { new Claim(ClaimTypes.NameIdentifier, "name") }));
+
+            // would block if EndPoint was executed
+            await dispatcher.ExecuteAsync<BlockingEndPoint>("", context).OrTimeout();
+
+            Assert.Equal(StatusCodes.Status401Unauthorized, context.Response.StatusCode);
+        }
+
+        private class TestAuthenticationHandler : IAuthenticationHandler
+        {
+            private readonly HttpContext Context;
```

Comment:
So many contexts that I don't know which context this context is in (contextually). Rename this to `HttpContext` or something.